### PR TITLE
Adjust `amp-list` container height for both load-end and load-failed elements

### PR DESCRIPTION
**Pull request by cathyxz**
Since it's sometimes the case that the `load-more-button`, `load-more-failed`, and `load-more-end` elements will have different height.

### CODE REVIEW

File: extensions/amp-list/0.1/amp-list.js [hunk 1]
```diff
@@ -707,47 +707,53 @@ export class AmpList extends AMP.BaseElement {
...
-      const targetHeight = target./*OK*/scrollHeight;
-      const height = this.element./*OK*/offsetHeight;
-      this.loadMoreEnabledPromise_.then(enabled => {
-        if (enabled) {
-          this.attemptToFitLoadMoreElements_(targetHeight, height);
-        } else {
+    this.loadMoreEnabledPromise_.then(enabled => {
+      if (enabled) {
+        const element = !!this.loadMoreSrc_ ?
+          this.loadMoreService_.getLoadMoreButton() :
```
Comment (cathyxz) [inline extensions/amp-list/0.1/amp-list.js]: Basically we want to adjust for the fact that load-more-button and load-more-end might not have the same height.

File: extensions/amp-list/0.1/amp-list.js [hunk 2]
```diff
@@ -892,14 +898,18 @@ export class AmpList extends AMP.BaseElement {
...
-            this.loadMoreService_.setLoadMoreFailed();
-          });
-          const loadMoreFailedClickable = this.loadMoreService_
-              .getLoadMoreFailedClickable();
-          this.unlistenLoadMore_ = listen(
-              loadMoreFailedClickable,
-              'click', () => this.loadMoreCallback_(/*opt_reload*/ true));
+          this.mutateElement(() => this.loadMoreService_.setLoadMoreFailed())
+              .then(() => {
+                this.attemptToFitLoadMoreElement_(
```
Comment (cathyxz) [inline extensions/amp-list/0.1/amp-list.js]: Here we want to do the same for `load-more-failed`.

Comment (cvializ) [inline extensions/amp-list/0.1/amp-list.js]: There are two `then()` blocks here. The first block should either return a `Promise` if the second block needs to wait for something to resolve, or the two blocks should be combined.

Comment (cathyxz) [reply extensions/amp-list/0.1/amp-list.js]: Yes. Great catch. Thank you!

Comment (cathyxz) [reply extensions/amp-list/0.1/amp-list.js]: Fixed this.

**Pull request merged**